src/window.cpp

```diff
@@ -0,0 +1,159 @@
+                     this->parent,                  /* parent window */
+                     this->x, this->y,              /* x, y */
+                     this->width, this->height,     /* width, height */
+                     this->b_width,                 /* border_width */
+                     XCB_WINDOW_CLASS_INPUT_OUTPUT, /* class */
+                     window::screen_->root_visual,  /* visual */
+                     mask, values);                 /* masks, not used yet */
+
+  /* Fixes window's place */
+  xcb_set_input_focus (c_, XCB_INPUT_FOCUS_POINTER_ROOT, id,
+    XCB_TIME_CURRENT_TIME); 
+  xcb_change_window_attributes (c_, id, XCB_CW_OVERRIDE_REDIRECT,
+    &override_redirect);
+  
+  /* Map the window on the screen */
+  xcb_map_window (window::c_, this->id);
+
+  /* Sends commands to the server */
+  xcb_flush (window::c_);
+};
+
+int
+window::get_screen_position (int DesktopNumber)
+ {
+   // As all screenshots have at least one common coordinate of corner, we need to find only the second one
+   int coord = k_dexpo_padding;
+   for (const auto &dexpo_pixmap : storage)
+     {
+       // Estimating all space before the screenshot we search
+       if (dexpo_pixmap.desktop_number < DesktopNumber)
+         {
+           coord += k_dexpo_padding;
+           if (k_dexpo_height == 0)
+             {
+               coord += dexpo_pixmap.height;
+             }
+           else if (k_dexpo_width == 0)
+             {
+               coord += dexpo_pixmap.width;
+             }
+         }
+       else
+         {
+           return coord;
+         }
+     }
+ }
+
+void window::highlight_window (int DesktopNumber) 
+{
+  int16_t x = k_dexpo_padding;
+  int16_t y = k_dexpo_padding;
+  uint32_t values[1]; // mask for changing border's color
+
+  if (k_dexpo_height == 0) 
+    {
+      y = get_screen_position (DesktopNumber);
+    }
+  else 
+    {
+      x = get_screen_position (DesktopNumber);
+    }
+  
+  // We can't control rectangle's thickness, so instead we spawn several close to each other
+  xcb_rectangle_t rectangles[] = {
+    {x-k_dexpo_hwidth, y-k_dexpo_hwidth, k_dexpo_hwidth, storage[DesktopNumber].height+2*k_dexpo_hwidth},                       // left border
+    {x+storage[DesktopNumber].width, y-k_dexpo_hwidth, k_dexpo_hwidth, storage[DesktopNumber].height+2*k_dexpo_hwidth},  // right border
+    {x-1, y-k_dexpo_hwidth, storage[DesktopNumber].width+1, k_dexpo_hwidth},                        // top border
+    {x-1, y+storage[DesktopNumber].height, storage[DesktopNumber].width+1, k_dexpo_hwidth}   // bottom border
+  };
+  // Drawing the highlighting itself
+  xcb_poly_fill_rectangle(window::c_, this->id, desktop_pixmap::gc_, 4, rectangles);
+
+  // Changing color
+  uint32_t mask = XCB_GC_FOREGROUND;
+  values[0] = k_dexpo_hcolor;
+  xcb_change_gc (c_, desktop_pixmap::gc_, mask, &values);
+}
+
+void window::remove_highlight (int DesktopNumber) 
```

Comment:
Change `DesktopNumber` case to snake_case

Reply:
Don't write the same function twice.
Just use `window::highlight_window` and pass it a highlight color.
This way you can pass it `k_dexpo_hcolor` when you want to highlight the window and
`k_dexpo_bcolor` when you want to remove the highlight